sympy/printing/codeprinter.py

```diff
@@ -14,6 +16,32 @@
 from sympy.codegen.ast import Assignment
 
 
+def prints_statement(method):
+    """ Decorator for methods printing statements (as opposed to expressions) """
+
+    @wraps(method)
+    def wrapper(self, expr):
+        if self._settings.get('enable_statements', True):
+            return method(self, expr)
+        else:
+            raise ValueError("The code-printer setting 'enable_statements' is False,\n"
+                             "but %s generates a statement in the chosen language." %
+                             type(expr))
```

Comment:
That's true, but it's a decorator that will be used in respective printer (would we ever want to print an expression as a statement? -- only use case I can think of is in try/except blocks and then enable_statements will be `True` any way. I guess we could have an option called "only_expressions" or something instead? I find "enable_statements" to be clearer though (minus the fact that an expression could also be a statement).